src/pages/resultsView/enrichments/ExpressionEnrichmentsContainer.tsx

```diff
@@ -108,43 +94,81 @@ export default class ExpressionEnrichmentContainer extends React.Component<IExpr
     }
 
     private dataStore = new EnrichmentsTableDataStore(
-        ()=>{
+        () => {
             return this.filteredData;
         },
-        ()=>{
+        () => {
             return this.highlightedRow;
         },
-        (c:ExpressionEnrichmentRow)=>{
+        (c: ExpressionEnrichmentRow) => {
             this.highlightedRow = c;
         }
     );
 
-    @computed get volcanoPlotLabels() {
-        if (this.props.alteredVsUnalteredMode) {
-            return ["Under-expressed", "Over-expressed"];
-        } else {
-            return [this.props.group2Name!, this.props.group1Name!];
-        }
+    //used in 2 groups analysis
+    @computed get group1() {
+        return this.props.groups[0];
     }
 
-    @computed get group1CheckboxLabel() {
-        if (this.props.alteredVsUnalteredMode) {
-            return "Over-expressed";
-        } else {
-            return <span style={{display:"flex", alignItems:"center"}}>Enriched in&nbsp;<EllipsisTextTooltip text={this.props.group1Name!} shownWidth={100}/></span>;
-        }
+    //used in 2 groups analysis
+    @computed get group2() {
+        return this.props.groups[1];
     }
 
-    @computed get group2CheckboxLabel() {
-        if (this.props.alteredVsUnalteredMode) {
-            return "Under-expressed";
+    @computed get selectedGenesSet() {
+        return _.keyBy(this.selectedGenes || []);
+    }
+
+    @computed get isTwoGroupAnalysis(): boolean {
+        return this.props.groups.length == 2;
+    }
+
+    @computed get customColumns() {
+        return getExpressionEnrichmentColumns(this.props.groups, this.props.alteredVsUnalteredMode);
+    }
+
+    @computed get visibleOrderedColumnNames() {
+        const columns = [];
+        columns.push(ExpressionEnrichmentTableColumnType.GENE,
+            ExpressionEnrichmentTableColumnType.CYTOBAND);
+
+        this.props.groups.forEach(group => {
+            columns.push(group.name + " mean");
```

Comment:
mean and standard deviation I think should be wrapped in a function that generates column name, or use an enum value like `ExpressionEnrichmentTableColumnType.MEAN_SUFFIX`